go/vt/vtgate/planbuilder/operators/aggregation_pushing.go

```diff
@@ -660,7 +722,11 @@ func (ab *aggBuilder) buildProjectionForAggr(lhsAE *sqlparser.AliasedExpr, rhsAE
 		As:   sqlparser.NewIdentifierCI(aggr.Original.ColumnName()),
 	}
 
-	ab.proj.addUnexploredExpr(projAE, projExpr)
+	_, err := ab.proj.addUnexploredExpr(projAE, projExpr)
+	if err != nil {
+		return nil
+	}
+	return err
```

Comment:
this looks wrong. If err != nil, we are returning nil, but otherwise we return err (which is now guaranteed to be nil).
Replace the whole thing with `return err`, I think